### PR TITLE
feat(compas-convert): get tests running, fix some issues while running the tests

### PR DESCRIPTION
**Pull request by dirkdev98**
See inline comments

### CODE REVIEW

File: packages/compas-convert/src/passes/add-common-imports.ts [hunk 1]
```diff
@@ -15,49 +15,52 @@ export function addCommonImports(context: Context, sourceFile: SourceFile) {
 		return;
 	}
 
-	addNamedImportIfNotExists(
+	addPendingImport(
```
Comment (dirkdev98) [inline packages/compas-convert/src/passes/add-common-imports.ts:18]: Had issues with saving imports, so migrated them all to this.

File: packages/compas-convert/src/passes/convert-test-config.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	const program = getTypescriptProgram(context);
+	const configFile = program.getSourceFile(configFilePath);
+
+	if (configFile) {
+		configFile.replaceWithText(`
+import { readFileSync } from "node:fs";
+import { parseEnv } from "node:util";
+import { refreshEnvironmentCache, loggerSetGlobalDestination } from "@compas/stdlib";
+
+await setupTestEnvironment();
```
Comment (dirkdev98) [inline packages/compas-convert/src/passes/convert-test-config.ts:20]: Setup files should auto-run, so this automatically injects everything when imported.

File: packages/compas-convert/src/passes/convert-test-files.ts [hunk 3]
```diff
@@ -47,7 +50,11 @@ export function convertTestFiles(context: Context, sourceFile: SourceFile) {
 		?.remove();
 
 	// re-add import for newTestEvent
-	addNamedImportIfNotExists(sourceFile, "@compas/cli", "newTestEvent", false);
+	addPendingImport(context, sourceFile, "@compas/cli", "newTestEvent", false);
```
Comment (dirkdev98) [inline packages/compas-convert/src/passes/convert-test-files.ts:53]: Same here with import issues, just decided to import everything always.

File: packages/compas-convert/src/passes/convert-test-files.ts [hunk 4]
```diff
@@ -128,7 +132,7 @@ function handleNestedTest(expression: CallExpression) {
 				const expected = args[1]?.getText() ?? "true";
 				const message = args[2]?.getText() ?? "";
 				it.expression = it.expression.replaceWithText(
-					`expect(${actual}${message ? `, ${message}` : ""}).toStrictEqual(${expected})`,
+					`expect(${actual}${message ? `, ${message}` : ""}).toEqual(${expected})`,
```
Comment (dirkdev98) [inline packages/compas-convert/src/passes/convert-test-files.ts:135]: Strict equal might be too strict? This is less strict on `undefined` vs missing keys for example.

File: packages/compas-convert/src/passes/convert-test-files.ts [hunk 5]
```diff
@@ -187,29 +184,13 @@ function handleNestedTest(expression: CallExpression) {
...
-		addNamedImportIfNotExists(expression.getSourceFile(), "vitest", "expect", false);
-	}
 }
 
 /**
  * Check if the expression only contains other tests
  */
 function testIsParent(expression: CallExpression): boolean {
-	return getDirectDescendants(expression).every((child) =>
+	return getDirectDescendants(expression).some((child) =>
```
Comment (dirkdev98) [inline packages/compas-convert/src/passes/convert-test-files.ts:193]: A `t.test()` should also become a `describe` when it has call expressions that aren't necessarily child `t.test`

File: packages/compas-convert/src/passes/init-typescript-in-project.ts [hunk 6]
```diff
@@ -49,10 +49,10 @@ export async function initTypescriptInProject(context: Context) {
 
 export default defineConfig({
   test: {
-  	globalSetup: "src/test/config.ts",
+    setupFiles: "./test/config.ts",
```
Comment (dirkdev98) [inline packages/compas-convert/src/passes/init-typescript-in-project.ts:52]: Setup files are loaded in the correct context, which allows es module live binding sharing.

File: packages/compas-convert/src/shared/import.ts
Comment (dirkdev98) [inline packages/compas-convert/src/shared/import.ts:1]: We had some duplicate functions.

**Review by jurriaanr — APPROVED** [review]

**Pull request merged**